### ISSUE TITLE
I wanna ask that How to modify the code to train 3D images?

### ISSUE THREAD

**Issue by InvincibleXiao** [Andrew-Qibin/CoordAttention#3]

**Comment by Andrew-Qibin** [comment]
Do you mean point cloud data?

**Comment by InvincibleXiao** [comment]
> Do you mean point cloud data?

I mean 3D medical images。

**Comment by InvincibleXiao** [comment]
> Do you mean point cloud data?

I made  some modify.Is that work?
class CoordAtt3D(nn.Module):
    def __init__(self, inp, oup, reduction=32):
        super(CoordAtt, self).__init__()
        self.pool_h = nn.AdaptiveAvgPool3d((None, 1, None))
        self.pool_w = nn.AdaptiveAvgPool3d((1, None, None))
        self.pool_L = nn.AdaptiveAvgPool3d((None, None, 1))

        mip = max(8, inp // reduction)

        self.conv1 = nn.Conv3d(inp, mip, kernel_size=1, stride=1, padding=0)
        self.bn1 = nn.BatchNorm3d(mip)
        self.act = h_swish()

        self.conv_h = nn.Conv3d(mip, oup, kernel_size=1, stride=1, padding=0)
        self.conv_w = nn.Conv3d(mip, oup, kernel_size=1, stride=1, padding=0)
        self.conv_l = nn.Conv3d(mip, oup, kernel_size=1, stride=1, padding=0)

    def forward(self, x):
        identity = x

        n, c, h, w, l = x.size()
        x_h = self.pool_h(x)
        x_w = self.pool_w(x).permute(0, 1, 3, 2, 4)
        x_l= self.pool_w(x).permute(0, 1, 4, 3, 2)

        y = torch.cat([x_h, x_w, x_l], dim=2)
        y = self.conv1(y)
        y = self.bn1(y)
        y = self.act(y)

        x_h, x_w = torch.split(y, [h, w, l], dim=2)
        x_w = x_w.permute(0, 1, 3, 2, 4)
        x_l= x_l.permute(0, 1, 4, 3, 2)

        a_h = self.conv_h(x_h).sigmoid()
        a_w = self.conv_w(x_w).sigmoid()
        a_l=self.conv_l(x_l).sigmoid()

        out = identity * a_w * a_h * a_l

        return out

**Comment by Andrew-Qibin** [comment]
I think it is correct.

**Comment by InvincibleXiao** [comment]
> I think it is correct.

I am not sure about why we concat at the dim=2.If I concat the 3D images at dim=2 .Is that work?

**Comment by Andrew-Qibin** [comment]
It needs a try.

**Comment by yan4821567** [comment]
> It needs a try.

I also want to use it in 3D images. I imitated the code under the 2D images to rewrite it into the 3D. But the main problem I met is 【torch.cat()】

Assume I have a 3D dataset, the shape of input is [2,32,112,160,128], which indicates [batch,channel,height,width,depth].
And I use:
        `self.pool_h = nn.AdaptiveAvgPool3d((None, None,1))
        self.pool_w = nn.AdaptiveAvgPool3d((None,1,None))
        self.pool_d = nn.AdaptiveAvgPool3d((1,None,None))
        ......
        x_h = self.pool_h(x)
        x_w = self.pool_w(x).permute(0, 1, 2, 4, 3)
        x_d = self.pool_d(x).permute(0, 1, 4, 3, 2)
       `
Then I get:
        `x_h.shape == [2,32,112,160,1]   
         x_w.shape == [2,32,112,128,1]
         x_d.shape == [2,32,128,160,1]`

Now, I can use ` torch.cat([x_h,x_w],3)` to get a tensor like [2,32,112,288,1], but I don't know how to contact [x_h,x_w,x_d]. Can you give some advice?
